### ISSUE TITLE
Backend and infra refactor 

### ISSUE THREAD

**Issue by kowalski** [rumblefishdev/evm-debugger#168]
TL;DR: we're moving from Fargate to Lambda and we will be using SQS. 

Steps of this refactoring:

- [ ] Start with dynamoDb model that stores our data (`/packages/transaction-trace-lambda/src/AnalyzerDataRepository.ts`).
     * In DynamoDB table we want to store 2 types of information:
       * What txHashes have been looked up (this is what we have currently)
       * Events that happened during processing (we don't have this yet)
     * We want to use a pattern as described here https://www.alexdebrie.com/posts/dynamodb-single-table/. Long story short we want to continue using txHash as Primary Key. We want to now add a sorting key will have following values:
       * For a document we save upon request just return something like "#TRANSACTION"
       * For a document presenting event return something like "EVENT${epochTimestamp}"
     * Thanks to above structure, when we query all items for a given txHash we will get a list of documents, where the first one is always the initial entry and the following ones are the events. 
     * To know current state of processing it will suffice to just look at the last event for a given txHash. 
     
- [ ] With the above structure in place make changes in the infra and `packages/transaction-trace-lambda/src/LambdaEntrypoint.ts` (please change the name "lambda" do "api") to
  * Just get data from dynamo. If there is no entry in dynamo, make an entry and put message in SQS queue. The entry in dynamo really only needs:
    * txHash
    * chainId
    * We no longer need `taskArn` in here
    * status=pending
    * add timestamp (why there is no timestamp here?)
    * sortingKey=#TRANSACTION
  * If status is pending or there is no status (SQS didn't pick it up yet) just return response and don't start another task.
  * If status is failed start a new task and make a item with status pending and put a new message in SQS  
  * Don't interact with ECS in any way.
  * Don't interact with S3 in any way.
  * Don't build the S3 path in this layer. The path to result on S3 should be taken form success event (assuming last event status is success).
    
- [ ] Now we're ready to change `packages/transaction-trace-provider/` into lambda. 
  * At the moment things like txHash, chainId, hardhatForkingUrl, and transactionS3Bucket are all passed through as env variables. Changes:
    * txHash, chain, hardhatForkingUrl - should be in SQS message, so in `event.messages[0]`
    * transactionS3Bucket should be passed in lambda environemtn from infrastracture level
  * Before starting doing antyhing, reset the fork (https://hardhat.org/hardhat-network/docs/guides/forking-other-networks#resetting-the-fork) as it may have stale state from previous processing. Mind that this is where you set the forkingUrl. It cannot be passed in environment in hardhat.config.ts (as done currently), because subsequent requests may be for different chainId, so it has to be reset before each processing.
  * Make a status entry before (running) and after procesing (success).
  * The success event should include the s3Path to result
  

- [ ] Make a failure handler - which we will hook to dead-letter queue. This handler should just make a dynamodb event entry with status=failed

- [ ] Changes in infra. 
  * Define SQS queue for task and SQS queue for dead-letter. 
  * Using same zip bundle for both lambda handlers but different entrypoint.
  * Set maxConcurrency to 5.
  * Set lambda timeout to 15 minutes.
  * Set batchSize=1
